sdk/cosmosdb/cosmos/src/client/Item/Item.ts

```diff
@@ -197,17 +251,44 @@ export class Item {
         this.partitionKey,
       );
 
-      const path = getPathFromLink(this.url);
-      const id = getIdFromLink(this.url);
+      let path = getPathFromLink(this.url);
+      let id = getIdFromLink(this.url);
 
-      const response = await this.clientContext.delete<T>({
-        path,
-        resourceType: ResourceType.item,
-        resourceId: id,
-        options,
-        partitionKey: this.partitionKey,
-        diagnosticNode,
-      });
+      if (this.clientContext.enableEncryption) {
```

Comment:
sems like this code is getting repeated in all the API's, can we extract it out at one place?

Reply:
There are slight tweaks inside this if block for various APIs, we can extract out for 1-2 APIs but it would still remain in majority of them.